src/jdk.compiler/share/classes/com/sun/tools/javac/platform/JDKPlatformProvider.java

```diff
@@ -83,7 +83,14 @@ public Iterable<String> getSupportedPlatformNames() {
     }
 
     @Override
-    public PlatformDescription getPlatform(String platformName, String options) {
+    public PlatformDescription getPlatform(String platformName, String options) throws PlatformNotSupported {
+        if (Source.lookup(platformName) == null) {
+            throw new PlatformNotSupported();
+        }
+        return getPlatformTrusted(platformName);
+    }
+
+    public PlatformDescription getPlatformTrusted(String platformName) {
```

Comment:
I noticed that `getPlatform` was not throwing an exception if the release was not valid, which then later results in an NPE. I've added an explicit check here instead. The caller can then catch the exception.

Reply:
We probably can eliminate the `options` (which is a historical artifact), but I don't think it needs to be done here